lib/src/prover.rs

```diff
@@ -21,19 +24,37 @@ impl From<String> for ProverError {
 
 pub type ProverResult<T, E = ProverError> = core::result::Result<T, E>;
 pub type ProverConfig = serde_json::Value;
-pub type Proof = serde_json::Value;
+pub type ProofKey = (ChainId, B256, u8);
+
+#[derive(Debug, Serialize, ToSchema, Deserialize, Default)]
+/// The response body of a proof request.
+pub struct Proof {
```

Comment:
Can this combine with TaskDescriptor, seems both are key index

Reply:
They are similar, but this proof object is meant to be a return from a prover, whereas the TaskDescriptor doesn't contain any return values just the query details with which to get the proof value (either from db or generated)